libraries/provider_database_mysql_user.rb

```diff
@@ -84,6 +84,11 @@ def action_grant
         def exists?
           db.query("SELECT User,host from mysql.user WHERE User = '#{@new_resource.username}' AND host = '#{@new_resource.host}'").num_rows != 0
         end
+
+        def mysql_escape (s)
+          # having in mind that escaped string will be used within single quotes!
+          s.gsub(/'/,"''").gsub(/\\/,'\\\\\\')
```

Comment:
@abrindeyev, SQL character escaping is a sensitive issue to implement. So I think it's better delegate to the root library whenever possible to avoid bugs.

AFAIK this cookbook [already uses this gem](https://github.com/opscode-cookbooks/database/blob/master/libraries/provider_database_mysql.rb#L29), so no need to add any new dependency. Furthermore, the gem [uses the native `mysql_escape_string` method](https://github.com/luislavena/mysql-gem/blob/ad959ba2dca95c0e841c4aa3a2d8e7c3898974e6/ext/mysql_api/mysql.c#L290).

Just my opinion. I'm not the maintainer :smiley: